executor/aggregate.go

```diff
@@ -344,6 +349,15 @@ func (e *HashAggExec) initForUnparallelExec() {
 	}
 }
 
+func closeBaseExecutor(b *baseExecutor) {
+	if r := recover(); r != nil {
+		if str, ok := r.(string); ok && strings.Contains(str, memory.PanicMemoryExceed) {
+			terror.Log(b.Close())
+			panic(r)
+		}
```

Comment:
How about the panic that caused by other problems?

Reply:
Most of the panic is mocked by the failpoint code or setting `@@tidb_mem_quota_query` in the test code.

I'm not sure in the real scene what will cause the aggregate executor `Open` function panic. I have to say, my purpose is not fix a bug, I write this PR mainly because it blocks the merging of https://github.com/pingcap/tidb/pull/26241

This change is conservative, the upside of being conservative is that it will not introduce bugs unintentionally...

Reply:
OK, I've changed here, @xiongjiwei found a bug.